vmm-reference-simple-handler.md

```diff
@@ -0,0 +1,137 @@
+# [Simple Handler](./../src/devices/src/virtio/net/simple_handler.rs)
+
+`SimpleHandler` contains methods for handling Tx/Rx queues. This handler is generalized over the implementaton of queues (does not make any assumptions the way queues are implemented). 
+
+```rs
+pub struct SimpleHandler<M: GuestAddressSpace, S: SignalUsedQueue> {
+    pub driver_notify: S,
+    pub rxq: Queue<M>,
+    pub rxbuf_current: usize,
+    pub rxbuf: [u8; MAX_BUFFER_SIZE],
+    pub txq: Queue<M>,
+    pub txbuf: [u8; MAX_BUFFER_SIZE],
+    pub tap: Tap,
+}
+
+```
+
+__A brief overview of all the struct variables:__
+
+1. `driver_notify:`
+    
+    A SignalUsedQueue object to model the operation of signalling the driver about used events. It should implement a method named `signal_used_queue` that takes parameter `index` and notifies the driver about index. Index values used here in SimpleHandler are `RXQ_INDEX` (for receiving) and `TXQ_INDEX` (for transmitting).
+
+2. `rxq:`  Recieving queue of `GuestAddressSpace`
+3. `rxbuf_current:` Number of bytes in `rxbuf` 
+4. `rxbuf:` Recieving buffer of maximum size `MAX_BUFFER_SIZE`
+5. `txq:`  Transmit queue of `GuestAddressSpace`
+6. `txbuf:` Transmit buffer of maximum size `MAX_BUFFER_SIZE`
+7. `tap:` Object of Tap struct that wraps the file descriptor for the tap device so methods can run ioctls on the interface
```

Comment:
7. `tap:` Object of  the Tap struct that wraps the file descriptor for the tap device so methods can run ioctls on the interface.